src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestFile.cs

```diff
@@ -12,8 +12,12 @@ namespace NuGet.Packaging
 {
     public class ManifestFile
     {
-        private static readonly char[] _invalidTargetChars = Path.GetInvalidFileNameChars().Except(new[] { '\\', '/', '.' }).ToArray();
-        private static readonly char[] _invalidSourceCharacters = Path.GetInvalidPathChars();
+        private static readonly char[] _invalidTargetChars = PackageReferenceSet.ReferenceFileInvalidCharacters.Except(new[] { '\\', '/', '.' }).ToArray();
```

Comment:
'.' isn't in ReferenceFileInvalidCharacters so it doesn't need to be excluded here